### PR TITLE
BUG: Fix #13213 json_normalize() and non-ascii characters in keys

### LINKED ISSUE #13213: json_normalize() can't deal with non-ascii characters in unicode keys
Example code:

```python
import pandas
import json

testjson = u'''
[{"Ünicøde":0,"sub":{"A":1, "B":2}},
 {"Ünicøde":1,"sub":{"A":3, "B":4}}]
 '''.encode('utf8')
pd.io.json.json_normalize(json.loads(testjson))
```

Output:

```
Traceback (most recent call last):
  File "...lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 2885, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-12-f866f9c7ec7c>", line 5, in <module>
    pd.io.json.json_normalize(json.loads(testjson))
  File ".../lib/python2.7/site-packages/pandas/io/json.py", line 715, in json_normalize
    data = nested_to_record(data)
  File ".../lib/python2.7/site-packages/pandas/io/json.py", line 617, in nested_to_record
    newkey = str(k)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xdc' in position 0: ordinal not in range(128)

```

Expected output

```
   sub.A  sub.B  Ünicøde
0      1      2        0
1      3      4        1
```

The cause are probably
https://github.com/pydata/pandas/blob/master/pandas/io/json.py#L618
and https://github.com/pydata/pandas/blob/master/pandas/io/json.py#L620

Those lines seemingly were introduced to deal with numeric types, but fail when `k` is a Unicode object containing non-ascii characters.

It seems to be the same bug in principle as https://github.com/pydata/pandas/issues/13101

### PR DESCRIPTION
**Pull request by fmarczin**
- [ ] closes #13213 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_json.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
```
Comment (jreback) [inline pandas/tests/frame/test_json.py]: no, this can go (just the test), don't add a new file in ``pandas/io/tests/json/test_json_norm.py``

**Comment by jreback** [comment]
pls add a whatsnote

File: pandas/tests/frame/test_json.py (hunk 1 above)
Comment (fmarczin) [inline pandas/tests/frame/test_json.py]: Not sure I understand the second part, but I removed the encoding-line

**Comment by jreback** [comment]
move the test to where I indicated

**Comment by jreback** [comment]
thanks @fmarczin !

**Comment by fmarczin** [comment]
Thank you for taking care of this so fast!

**Pull request closed without merging**